SDWebImage/Core/SDImageCache.m

```diff
@@ -528,20 +528,18 @@ - (nullable NSOperation *)queryCacheOperationForKey:(nullable NSString *)key opt
         @autoreleasepool {
             NSData *diskData = [self diskImageDataBySearchingAllPathsForKey:key];
             UIImage *diskImage;
-            SDImageCacheType cacheType = SDImageCacheTypeNone;
             if (image) {
                 // the image is from in-memory cache, but need image data
                 diskImage = image;
-                cacheType = SDImageCacheTypeMemory;
```

Comment:
This behavior not sounds reasonable.
This line of code from https://github.com/SDWebImage/SDWebImage/pull/2218
However, this may cause misunderstanding. The image actually from disk and spending time on disk query, user may use that `cacheType` to detect whether this callback is async or sync.